src/os/ui/datetime/datecontrol.js

```diff
@@ -205,12 +219,32 @@ os.ui.datetime.DateControlCtrl.prototype.onDurationChanged = function() {
   if (!this['disabled']) {
     this['startDate'] = os.time.floor(this['startDate'], this['duration'], true);
 
-    if (this['duration'] === os.time.Duration.CUSTOM) {
-      // for custom duration, make dates the same (end is inclusive)
-      this['endDate'] = new Date(this['startDate']);
-    } else {
-      // for all other durations, set the end date from the start date
-      this['endDate'] = os.time.offset(this['startDate'], this['duration'], 1, true);
+    switch (this['duration']) {
+      case os.time.Duration.LAST24HOURS:
+        this.setRelativeDateRange(1);
+        break;
+      case os.time.Duration.LAST48HOURS:
+        this.setRelativeDateRange(2);
+        break;
+      case os.time.Duration.LAST7DAYS:
+        this.setRelativeDateRange(7);
+        break;
+      case os.time.Duration.LAST14DAYS:
+        this.setRelativeDateRange(14);
+        break;
+      case os.time.Duration.LAST30DAYS:
+        this.setRelativeDateRange(30);
+        break;
+      case os.time.Duration.CUSTOM:
+        this['relativeDuration'] = false;
```

Comment:
I would consider setting `this['relativeDuration'] = false;` prior to the `switch` statement and letting `setRelativeDateRange` change it back. That would avoid the need to add that line for any non-relative duration.